### PR TITLE
Issue #424: Add CLI options for specific query execution - SNMP

### PR DESCRIPTION
**Pull request by CherfaElyes**
* Added IQuery interface to handle protocols queries.
* Developed SnmpCli executable which enables to execute SNMP get, getNext & walk commands through CLI.
* Tested the generated SNMP.exe with different commands.

# Tests
## SNMP Get
image

 ## SNMP GetNext
image

## SNMP Walk
image

### CODE REVIEW

File: metricshub-agent/src/main/java/org/sentrysoftware/metricshub/cli/snmp/SnmpCli.java [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+import picocli.CommandLine;
+import picocli.CommandLine.ArgGroup;
+import picocli.CommandLine.Model.CommandSpec;
+import picocli.CommandLine.Option;
+import picocli.CommandLine.ParameterException;
+import picocli.CommandLine.Parameters;
+import picocli.CommandLine.Spec;
+
+@Data
+public class SnmpCli implements IQuery, Callable<Integer> {
```
Comment (NassimBtk) [inline metricshub-agent/src/main/java/org/sentrysoftware/metricshub/cli/snmp/SnmpCli.java:48]: Add Javadoc please.

File: metricshub-agent/src/main/java/org/sentrysoftware/metricshub/cli/snmp/SnmpCli.java [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+			})
+			.orElseThrow(() ->
+				new ParameterException(
+					spec.commandLine(),
+					"At least one SNMP query must be specified: --snmp-get, --snmp-getnext, --snmp-walk."
+				)
+			);
+	}
+
+	public static void main(String[] args) {
```
Comment (NassimBtk) [inline metricshub-agent/src/main/java/org/sentrysoftware/metricshub/cli/snmp/SnmpCli.java:124]: Javadoc.

File: metricshub-engine/src/main/java/org/sentrysoftware/metricshub/engine/common/IQuery.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
+package org.sentrysoftware.metricshub.engine.common;
+
+import com.fasterxml.jackson.databind.JsonNode;
```
Comment (NassimBtk) [inline metricshub-engine/src/main/java/org/sentrysoftware/metricshub/engine/common/IQuery.java:3]: Move this import after the MetricsHub Engine bloc.

File: metricshub-snmp-extension-common/src/main/java/org/sentrysoftware/metricshub/extension/snmp/AbstractSnmpRequestExecutor.java [hunk 4]
```diff
@@ -254,6 +256,37 @@ public enum SnmpGetRequest {
 		 * Represents an SNMP TABLE request.
 		 * Used to retrieve a table of SNMP objects.
 		 */
-		TABLE
+		TABLE,
+		/**
+		 *
```
Comment (NassimBtk) [inline metricshub-snmp-extension-common/src/main/java/org/sentrysoftware/metricshub/extension/snmp/AbstractSnmpRequestExecutor.java:261]: Add comment.

File: metricshub-snmp-extension/src/main/java/org/sentrysoftware/metricshub/extension/snmp/SnmpExtension.java [hunk 5]
```diff
@@ -97,4 +102,41 @@ public String getIdentifier() {
...
+			} else if (action.equals(GET_NEXT)) {
+				result = snmpRequestExecutor.executeSNMPGetNext(oId, snmpConfiguration, hostname, false);
+			} else if (action.equals(WALK)) {
+				result = snmpRequestExecutor.executeSNMPWalk(oId, snmpConfiguration, hostname, false);
+			} else {
+				throw new IllegalArgumentException(String.format("Hostname %s - Invalid SNMP Operation", hostname));
+			}
+
+			// Printing the result on the CLI
+			printWriter.print("\n\u001B[34m");
```
Comment (NassimBtk) [inline metricshub-snmp-extension/src/main/java/org/sentrysoftware/metricshub/extension/snmp/SnmpExtension.java:131]: Remove the color please.

**Review by NassimBtk — CHANGES_REQUESTED** [review]
Small updates and we are good to go.

File: metricshub-agent/src/main/java/org/sentrysoftware/metricshub/cli/snmp/SnmpCli.java [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+	@Option(names = "--snmp-get", order = 1, paramLabel = "OID", description = "SNMP Get request")
+	String get;
+
+	@Option(names = "--snmp-getnext", order = 2, paramLabel = "OID", description = "SNMP Get Next request")
+	String getNext;
+
+	@Option(names = "--snmp-walk", order = 3, paramLabel = "OID", description = "SNMP Walk request")
+	String walk;
+
+	@ArgGroup(exclusive = false, heading = "%n@|bold,underline SNMP Options|@:%n")
```
[the remark this answers is not in the document]
Comment (CherfaElyes) [reply metricshub-agent/src/main/java/org/sentrysoftware/metricshub/cli/snmp/SnmpCli.java:65]: Based on what I've noticed on MetricsHubCliService, order is not included in all ArgGroups 🤷‍♂️

File: metricshub-agent/src/main/java/org/sentrysoftware/metricshub/cli/snmp/SnmpCli.java [hunk 7]
```diff
@@ -0,0 +1,174 @@
...
+		return queryNode;
+	}
+
+	/**
+	 * Validates SNMP configuration and ensures exactly one query type (--snmp-get, --snmp-getnext, or --snmp-walk) is specified.
+	 *
+	 * @throws ParameterException if SNMP is not configured, no query is specified, or multiple queries are specified.
+	 */
+	void validate() throws ParameterException {
+		if (snmpConfigCli == null) {
```
[the remark this answers is not in the document]
Comment (CherfaElyes) [reply metricshub-agent/src/main/java/org/sentrysoftware/metricshub/cli/snmp/SnmpCli.java:103]: It's simple especially on SNMP, is it the same thing on the other protocols?

[the remark this answers is not in the document]
Comment (NassimBtk) [reply metricshub-agent/src/main/java/org/sentrysoftware/metricshub/cli/snmp/SnmpCli.java:103]: Each CLI is protocol-specific, so other protocols might not share the same behavior. For instance, in HTTP, you could imagine a command like `http.exe https://domain.net/systems` that execute an HTTP GET on the specified url.

**Pull request closed without merging**